dietstory-routing/game_asset_manager/views.py

```diff
@@ -94,33 +97,120 @@ def get(self, request, *args, **kwargs):
+                                    type: string
+                                    description: No version of the game exists.
+        """
+		try:
+			game_version = GameVersions.objects.get(is_live=True)
+		except GameVersions.DoesNotExist:
+			game_version = None
 
-        elif self.request.method == "POST":
-            return (permissions.IsAdminUser(),)
+		if game_version:
+			return JsonResponse(
+				{'game_version': GameVersionSerializer(game_version).data},
+				status=status.HTTP_200_OK)
+		else:
+			return JsonResponse(
+				{'message': 'No version of the game exists.'},
+				status=status.HTTP_404_NOT_FOUND)
 
-    def get(self, request, *args, **kwargs):
+	def post(self, request, *args, **kwargs):
+		"""
+        summary: Post Game Version
+        description: Submit game version with live by time.
+        tags:
+            - GameVersionView
+        responses:
+            200:
+                content:
+                    application/json:
+                        schema:
+                            type: object
+                            properties:
+                                message:
+                                    type: string
+                                    description: Game version has been submitted successfully.
+            400:
+                content:
+                    application/json:
+                        schema:
+                            type: object
+                            properties:
+                                message:
+                                    type: string
+                                    description: Invalid input parameters.
+            500:
+                content:
+                    application/json:
+                        schema:
+                            type: object
+                            properties:
+                                message:
+                                    type: string
+                                    description: Game version submission was not successful.
+        """
+		game_version_form = SubmitGameVersionForm(request.data)
 
-        try:
-            game_version = GameVersions.objects.get(is_live=True)
-        except GameVersions.DoesNotExist:
-            game_version = None
+		if game_version_form.is_valid():
 
-        if game_version:
-            return JsonResponse(
-                {'game_version': GameVersionSerializer(game_version).data},
-                status=status.HTTP_200_OK)
-        else:
-            return JsonResponse(
-                {'message': 'No version of the game exists.'},
-                status=status.HTTP_404_NOT_FOUND)
+			try:
+				major_ver = game_version_form.cleaned_data.get('major_ver')
+				minor_ver = game_version_form.cleaned_data.get('minor_ver')
+				live_by = game_version_form.cleaned_data.get('live_by')
+
+				game_version = GameVersions(major_ver=major_ver, minor_ver=minor_ver, live_by=live_by)
+				game_version.save()
+
```

Comment:
save only after scheduler adds the job

Reply:
Not sure I agree with this one

Reply:
We may run into an issue where the schedule job is finished before it is saved?

Reply:
well consider the issues in either case:
- game version saved but no schedule was created (saving before job creation)
- scheduler ran before game_version saved (saving after job creation)

In the first case, we have an implicit failure, and likely unnoticed until after version live failed to change.

In the second case, we get an immediate failure with no side effects.

The scheduler should gracefully fail if the version doesn't exist.